src/state-summary/state-card-input_text.html

```diff
@@ -0,0 +1,108 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+
+<link rel="import" href="../../bower_components/iron-flex-layout/iron-flex-layout-classes.html">
+
+<link rel="import" href="../../bower_components/paper-input/paper-input.html">
+
+<link rel="import" href="../components/entity/state-info.html">
+
+<link rel="import" href="../util/hass-util.html">
+
+<dom-module id="state-card-input_text">
+  <template>
+    <style is="custom-style" include="iron-flex iron-flex-alignment"></style>
+    <style>
+      paper-input {
+        margin-left: 16px;
+      }
+    </style>
+
+    <div class='horizontal justified layout'>
+      <state-info state-obj="[[stateObj]]" in-dialog='[[inDialog]]'></state-info>
+      <paper-input
+        minlength='[[min]]' maxlength='[[max]]' value='{{value}}' pin
+        disabled='[[disabled]]' auto-validate pattern='[[pattern]]'
+        on-change='selectedValueChanged' on-tap='stopPropagation'>
+      </paper-input>
+    </div>
+  </template>
+</dom-module>
+
+<script>
+class StateCardInputText extends Polymer.Element {
+
+  static get is() { return 'state-card-input_text'; }
+
+  static get properties() {
+    return {
+      hass: Object,
+
+      inDialog: {
+        type: Boolean,
+        value: false,
+      },
+
+      stateObj: {
+        type: Object,
+        observer: 'stateObjectChanged',
+      },
+
+      min: {
+        type: Number,
+        value: 0,
+      },
+
+      max: {
+        type: Number,
+        value: 100,
+      },
+
+      pattern: {
+        type: String,
+        value: '.*',
+      },
+
+      disabled: {
+        type: Boolean,
+        value: false,
+      },
+
+      value: {
+        type: String,
+      }
+
+    };
+  }
+
+  static get observers() {
+    return [
+    ];
+  }
+
+  stateObjectChanged(newVal) {
+    this.min = Number(newVal.attributes.min);
```

Comment:
No need to cast to number, they should be numbers coming in from HA. If that's not the case, we should update the component.

Reply:
I'll remove these redundant casts.